# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution

In [54]:
import pandas as pd
import numpy as np
from scipy import stats
import statsmodels.stats.api as sms
from math import sqrt

In [5]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [25]:
call=data.call
white = call[data.race=='w']
black = call[data.race=='b']
tcall=sum(call)

392.0

In [47]:
# number of callbacks for black-sounding names
wcall=sum(data[data.race=='w'].call)

In [48]:
bcall=sum(data[data.race=='b'].call)

<div class="span5 alert alert-success">
<p>Q1. since the data is comparing 2 sample, the test is better with two-sample t-test.Also there is 397 sample, which is large, therefore this apply to CLT. </p>
<p>Q2. null hypothesis: There is no difference between call back for balck sounding names and white sounding call names</p>
</div>


In [45]:
w = call[data.race=='w']
b = call[data.race=='b']
p=stats.ttest_ind(w, b, equal_var=True)
print("The t score is %.3f, the p-value is %.7f." % p)

The t score is 4.115, the p-value is 0.0000394.


In [79]:
# Calculate the mean difference and 95% confidence interval
cm = sms.CompareMeans(sms.DescrStatsW(w), sms.DescrStatsW(b))
print(cm.tconfint_diff(usevar='unequal'))

(0.016770673983991798, 0.04729503443489937)


<div class="span5 alert alert-success">
<p>Q3. the confidence interval of mean difference is 0.0167 and 0.0452. and p value is 0.0000394 which is close to 0.
</p>
</div>


<div class="span5 alert alert-success">
<p>Q4. the p-value is less than 0.05, close to 0. THis means the null hypothesis is rejected and there is significant difference between black sounding name call back and white sounding names. Therefore, there is racial discrimination in job market.<p>
<p>Q5. The analysis between name and call back shows there is significant difference between number of call back depend on their name sound. However, it does not mean the name is most important factor for call back. There are many other variables,haven't been analysed yet,that might be have more important and significant impact on call back. To identify which of the variables is important variable, we can test correlation test or linear regression test.  <p>    
</div>